src/i_flmusic.c

```diff
@@ -0,0 +1,288 @@
+                              fsynth_reverb_roomsize);
+        fluid_settings_setnum(settings, "synth.reverb.damp",
+                              fsynth_reverb_damp);
+        fluid_settings_setnum(settings, "synth.reverb.width",
+                              fsynth_reverb_width);
+        fluid_settings_setnum(settings, "synth.reverb.level",
+                              fsynth_reverb_level);
+    }
+
+    if (fsynth_chorus_active)
+    {
+        fluid_settings_setnum(settings, "synth.chorus.level",
+                              fsynth_chorus_level);
+        fluid_settings_setnum(settings, "synth.chorus.depth",
+                              fsynth_chorus_depth);
+        fluid_settings_setint(settings, "synth.chorus.nr", fsynth_chorus_nr);
+        fluid_settings_setnum(settings, "synth.chorus.speed",
+                              fsynth_chorus_speed);
+    }
+
+    synth = new_fluid_synth(settings);
+
+    sf_id = fluid_synth_sfload(synth, fsynth_sf_path, true);
+    if (sf_id == FLUID_FAILED)
+    {
+        // deleting the synth also deletes sfloader
+        delete_fluid_synth(synth);
+        delete_fluid_settings(settings);
+        fprintf(stderr,
+                "I_FL_InitMusic: Error loading FluidSynth soundfont: '%s'.\n",
+                fsynth_sf_path);
+        return false;
+    }
+
+    printf("I_FL_InitMusic: Using '%s'.\n", fsynth_sf_path);
+
+    return true;
+}
+
+static void I_FL_SetMusicVolume(int volume)
+{
+    // FluidSynth's default is 0.2. Make 1.0 the maximum.
+    // 0 -- 0.2 -- 10.0
+    fluid_synth_set_gain(synth, (float) volume / 127);
+}
+
+static void I_FL_PauseSong(void)
+{
+    fluid_player_stop(player);
+}
+
+static void I_FL_ResumeSong(void)
+{
+    fluid_player_play(player);
+}
+
+static void I_FL_PlaySong(void *handle, boolean looping)
+{
+    fluid_player_set_loop(player, looping ? -1 : 1);
+    fluid_player_play(player);
+}
+
+static void I_FL_StopSong(void)
+{
+    if (player)
+    {
+        fluid_player_stop(player);
+    }
+}
+
+// Determine whether memory block is a .mid file
+
+static boolean IsMid(byte *mem, int len)
+{
+    return len > 4 && !memcmp(mem, "MThd", 4);
+}
+
+static void *I_FL_RegisterSong(void *data, int len)
+{
+    int result = 0;
+
+    player = new_fluid_player(synth);
+
+    if (IsMid(data, len))
+    {
+        result = fluid_player_add_mem(player, data, len);
+    }
+    else
+    {
+        // Assume a MUS file and try to convert
+        MEMFILE *instream;
+        MEMFILE *outstream;
+        void *outbuf;
+        size_t outbuf_len;
+
+        instream = mem_fopen_read(data, len);
+        outstream = mem_fopen_write();
+
+        if (mus2mid(instream, outstream) == 0)
+        {
+            mem_get_buf(outstream, &outbuf, &outbuf_len);
+            result = fluid_player_add_mem(player, outbuf, outbuf_len);
+        }
+
+        mem_fclose(instream);
+        mem_fclose(outstream);
+    }
+
+    if (result != FLUID_OK)
```

Comment:
If `IsMid` returns false and `mus2mid` fails `result` will be left to 0 which is `FLUID_OK`. This is probably not what you wanted.

Reply:
Yep, good catch!